modules/Hoot/ui/sidebar/advancedOpts.js

```diff
@@ -646,4 +744,132 @@ export default class AdvancedOpts {
 
         return options;
     }
+
+    favoriteCheck(favorite) {
+        let type = d3.select( '#conflateType' ).property( 'value' ).toLowerCase();
+        if ( type === favorite[0].name ) {
+            return 'favopt';
+        }
+        else {
+            return 'text-input';
+        }
+    }
+
+    updateFavoriteOpts( toUpdate ) {
+
+        let getMem = [];
+
+        toUpdate.members.forEach( function(m) { getMem.push( m ); } );
+
+        let updateOpts = [];
+
+        function flatten( arr ) {
+            return arr.reduce( function( flat, toFlatten) {
+                return flat.concat( Array.isArray(toFlatten) ?
+                flatten(toFlatten)
+                : checkType(toFlatten) );
+            }, []);
+        }
+
+        function checkType( member ) {
+            if ( member.input === 'checkbox' ) {
+                    updateOpts.push(
+                        {
+                            input: member.input,
+                            default: d3.select(`#${member.id}`).select('input').property('checked'),
+                            id: member.id,
+                            description: member.description,
+                            label: member.label,
+                            type: member.type,
+
+                        }
+                    );
+            }
+            else {
+                    updateOpts.push(
+                        {
+                            input: member.input,
+                            default: d3.select(`#${member.id}`).select('input').property('value'),
+                            id: member.id,
+                            description: member.description,
+                            label: member.label,
+                            type: member.type,
+
+                        }
+                    );
+            }
+        }
+
+        flatten(getMem);
+
+        return updateOpts;
+    }
+
+    savingFavoriteOpts( opt ) {
+
+        let getAdvOptMembers = [];
+
+        let getFavOptMembers = [];
```

Comment:
minor nitpick, remove these 2 empty lines just so declared variables at top of function are grouped up